apps/api/src/app/environments/usecases/construct-framework-workflow/construct-framework-workflow.usecase.ts

```
@@ -178,4 +183,34 @@ export class ConstructFrameworkWorkflow {
       skip: (controlValues) => false,
     };
   }
+  private async getDbWorkflow(environmentId: string, workflowId: string): Promise<NotificationTemplateEntity> {
+    const foundWorkflow = await this.workflowsRepository.findByTriggerIdentifier(environmentId, workflowId);
+
+    if (!foundWorkflow) {
+      throw new InternalServerErrorException(`Workflow ${workflowId} not found`);
```

This should either throw a NotFound exception or return null and let the caller of the use-case handle null.

---

@rifont it's causing a problematic error